lib/elixir/lib/enum.ex

```diff
@@ -2573,9 +2604,17 @@ defmodule Enum do
 
   ## fetch
 
-  defp do_fetch([h | _], 0), do: {:ok, h}
-  defp do_fetch([_ | t], n), do: do_fetch(t, n - 1)
-  defp do_fetch([], _),      do: :error
+  @spec fetch_list(list, non_neg_integer) :: {:ok, element} | :error
+  defp fetch_list([], _index),
+    do: :error
+  defp fetch_list([elem], 0),
+    do: {:ok, elem}
+  defp fetch_list([_], _index),
+    do: :error
+  defp fetch_list([h | _], 0),
```

Comment:
Since we touching this place we can get rig of one-letter variable and use `head` and `tail`.